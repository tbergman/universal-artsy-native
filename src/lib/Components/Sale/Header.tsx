import React from "react"
import { Dimensions, StyleSheet, TextStyle, View } from "react-native"
import { createFragmentContainer, graphql } from "react-relay"
import OpaqueImageView from "../OpaqueImageView"
import Headline from "../Text/Headline"

import { Header_sale } from "__generated__/Header_sale.graphql"

interface Props {
  sale: Header_sale
  showImage: boolean
}

class Header extends React.Component<Props> {
  render() {
    const { width, height } = Dimensions.get("window")
    const sale = this.props.sale
    return (
      <View>
        <View style={styles.header}>
          {this.props.showImage ? (
            <OpaqueImageView
              // TODO: un-hardcode this (metaphysics isn't resolving sales' cover_image field property.)
              imageURL="https://d32dm0rphc51dk.cloudfront.net/3YueUJ2y1Vwx-YQJ82lG_w/wide.jpg"
              style={{ position: "absolute", top: 0, width, height }}
            />
          ) : null}
          <Headline style={styles.headline} numberOfLines={2}>
            {sale.name}
          </Headline>
        </View>
      </View>
    )
  }
}

export default createFragmentContainer(
  Header,
  graphql`
    fragment Header_sale on Sale {
      name
      cover_image {
        href
      }
    }
  `
)

const isPad = Dimensions.get("window").width > 700

interface Styles {
  header: TextStyle
  headline: TextStyle
}

const styles = StyleSheet.create<Styles>({
  header: {
    marginTop: 15,
    height: 36,
    justifyContent: "center",
    alignItems: "center",
    paddingLeft: 40,
    paddingRight: 40,
  },
  headline: {
    textAlign: "center",
    fontSize: isPad ? 20 : 14,
  },
})
