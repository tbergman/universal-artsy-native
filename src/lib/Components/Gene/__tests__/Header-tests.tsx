import { NativeModules } from "react-native"

import React from "react"
import * as renderer from "react-test-renderer"

import Header from "../Header"

beforeAll(() => {
  NativeModules.ARTemporaryAPIModule = { followStatusForGene: jest.fn() }
})

it("renders properly", () => {
  const gene = {
    __id: "gene-deep-time",
    _id: "gravity-id",
    id: "deep-time",
    name: "Deep Time",
  }

  const header = renderer.create(<Header gene={gene} shortForm={false} />).toJSON()
  expect(header).toMatchSnapshot()
})
