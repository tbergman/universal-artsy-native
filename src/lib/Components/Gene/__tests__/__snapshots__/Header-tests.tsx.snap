// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<Header
  gene={
    Object {
      "__id": "gene-deep-time",
      "_id": "gravity-id",
      "id": "deep-time",
      "name": "Deep Time",
    }
  }
  shortForm={false}
/>
`;
